src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -744,7 +749,17 @@ private bool TakeStartLine(SocketInput input)
                 RequestUri = requestUrlPath;
                 QueryString = queryString;
                 HttpVersion = httpVersion;
-                Path = RequestUri;
+
+                if (!string.IsNullOrEmpty(_pathBase) && requestUrlPath.StartsWith(_pathBase))
```

Comment:
case insensitive compare, then make sure you set PathBase to the requestUrlPath substring to preserve the user casing.

Also, StartsWith isn't good enough. "/foobar" starts with "/foo".